packages/server/graphql/mutations/addTeam.ts

```diff
@@ -62,9 +64,11 @@ export default {
         }
         return standardError(new Error('Failed input validation'), {userId: viewerId})
       }
-      if (orgTeams.length >= Threshold.MAX_FREE_TEAMS) {
-        const organization = await dataLoader.get('organizations').load(orgId)
-        const {tier}: {tier: TierEnum} = organization
+      const {tier, featureFlags} = organization
+      const maxTeams = featureFlags?.includes('teamsLimit')
+        ? Threshold.MAX_PERSONAL_TIER_TEAMS
+        : Threshold.MAX_FREE_TEAMS
+      if (orgTeams.length >= maxTeams) {
```

Comment:
I don't think this is a correct validation, because for the limit we count only **sticky teams** and restrict account only after it was flagged. 
i.e. `orgTeams.length >= maxTeams` could be true, but the limit might not be exceeded

I think we just need to check here for `limitExceededAt` flag

Checking for feature flag becomes redundant in this case